src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -146,6 +148,19 @@ internal sealed class ClassEvaluator : MemberEvaluator {
             }
         }
 
+        /// <summary>
+        /// Decides if this class is an abstract class or not
+        /// </summary>
+        private void DecideAbstract() {
+            // A class is derived from an abstract class if one of its bases is abstract or one of its bases is also derived from an abstract class
+            var derivedFromAbstract = _class.Bases.OfType<PythonClassType>().Any(b => b.IsAbstract || b.IsDerivedFromAbstractClass);
```

Comment:
This might be a useful extension to `IPythonClassType` i.s. `IsAbstract(this ...)`

Reply:
I set isAbstract on the class type in order to prevent doing more computations each time someone references isAbstract. I wouldn't think it would be an extension but rather a property. Do you think that would be better than computing it once and setting it on the class type? It would technically always be correct even if someone added new bases or changed methods.